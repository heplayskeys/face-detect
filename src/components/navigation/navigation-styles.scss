$breakpoint-phone: 450px;

.sign-out {
	@media screen and (max-width: $breakpoint-phone) {
		font-size: 1em;
		background-color: hsla(0, 0%, 100%, 0.4);
		border-top-left-radius: 0.5rem;
		border-bottom-left-radius: 0.5rem;
		border-color: rgba(0, 0, 0, 0.1);
		border-style: solid;
		border-width: 1px;
	}
}
