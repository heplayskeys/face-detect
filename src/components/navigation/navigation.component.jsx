import React from 'react';
import { connect } from 'react-redux';
import { updateRoute } from '../../redux/route/route.actions';
import AvatarIcon from '../avatar-icon/avatar-icon.component';
import './navigation.styles.scss';

export const Navigation = ({ route, setRoute, activeUser }) => {
	const displayRoute = {
		register: 'Sign In',
		signin: 'Register',
		home: 'Sign Out'
	};

	return (
		<nav className='flex justify-end'>
			{activeUser ? (
				<AvatarIcon />
			) : (
				<p
					className='sign-out link dim black underline pa3 pointer'
					onClick={() => setRoute(route)}
				>
					{displayRoute[route]}
				</p>
			)}
		</nav>
	);
};

const mapStateToProps = state => ({
	route: state.setRoute.route,
	activeUser: state.setUser.activeUser
});

const mapDispatchToProps = dispatch => ({
	setRoute: route => dispatch(updateRoute(route))
});

export default connect(mapStateToProps, mapDispatchToProps)(Navigation);
