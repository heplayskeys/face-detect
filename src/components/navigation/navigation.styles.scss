$breakpoint-phone: 450px;

.sign-out {
	font-size: 1.5rem;

	@media screen and (max-width: $breakpoint-phone) {
		font-size: 1rem;
		background-color: hsla(0, 0%, 100%, 0.4);
		border-radius: 0.5rem 0 0 0.5rem;
		border-color: rgba(0, 0, 0, 0.1);
		border-style: solid;
		border-width: 1px;
	}
}

.user-icon {
	padding: 1.5rem;

	@media screen and (max-width: $breakpoint-phone) {
		margin: 10px auto 0 10px;
		padding: 0;
	}
}
