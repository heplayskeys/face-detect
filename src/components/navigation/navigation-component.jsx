import React from 'react';
import { connect } from 'react-redux';
import { updateRoute } from '../../redux/route/route.actions';
import './navigation-styles.scss';

const Navigation = ({ route, setRoute, activeUser }) => {
	const displayRoute = {
		register: 'Sign In',
		signin: 'Register',
		home: 'Sign Out'
	};

	return (
		<nav className='flex justify-end'>
			{activeUser ? (
				<div className='tc grow pointer' style={{ padding: '1.5rem' }}>
					<p
						className='flex items-center justify-center br-100 pa3 m0 ba b--white h3 w3 f2 shadow-3 grow'
						style={{
							margin: 0,
							background: '#000',
							color: '#fff'
						}}
					>
						{activeUser.name[0].toUpperCase()}
					</p>
				</div>
			) : null}
			<p
				className='sign-out f3 link dim black underline pa3 pointer'
				onClick={() => setRoute(route)}
			>
				{displayRoute[route]}
			</p>
		</nav>
	);
};

const mapStateToProps = state => ({
	route: state.setRoute.route,
	activeUser: state.setUser.activeUser
});

const mapDispatchToProps = dispatch => ({
	setRoute: route => dispatch(updateRoute(route))
});

export default connect(mapStateToProps, mapDispatchToProps)(Navigation);
