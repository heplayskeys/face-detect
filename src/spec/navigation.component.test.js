import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Navigation } from '../components/navigation/navigation.component';

let wrapper;

describe('<Navigation />', () => {
	beforeEach(() => {
		const mockProps = {
			route: 'signin',
			setRoute: jest.fn(),
			activeUser: null
		};
		wrapper = shallow(<Navigation {...mockProps} />);
	});

	it('renders the navigation component successfully for signin', () => {
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});

	it('renders the navigation component successfully for register', () => {
		wrapper.setProps({ route: 'register' });
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});

	it('renders the navigation component successfully for home', () => {
		wrapper.setProps({ route: 'home', activeUser: { name: 'Enzyme' } });
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});
});
