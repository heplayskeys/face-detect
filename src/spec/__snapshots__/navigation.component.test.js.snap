// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> renders the navigation component successfully for home 1`] = `
<nav
  className="flex justify-end"
>
  <Connect(AvatarIcon) />
</nav>
`;

exports[`<Navigation /> renders the navigation component successfully for register 1`] = `
<nav
  className="flex justify-end"
>
  <p
    className="sign-out link dim black underline pa3 pointer"
    onClick={[Function]}
  >
    Sign In
  </p>
</nav>
`;

exports[`<Navigation /> renders the navigation component successfully for signin 1`] = `
<nav
  className="flex justify-end"
>
  <p
    className="sign-out link dim black underline pa3 pointer"
    onClick={[Function]}
  >
    Register
  </p>
</nav>
`;
